src/Adaptive.Agrona/Concurrent/UnsafeBuffer.cs

```diff
@@ -257,13 +258,7 @@ public void SetMemory(int index, int length, byte value)
 #endif
         {
             BoundsCheck0(index, length);
-
-            // TODO PERF Naive implementation, we should not write byte by byte, this is slow
-            //UNSAFE.SetMemory(byteArray, addressOffset + index, length, value);
-            for (var i = index; i < index + length; i++)
-            {
-                _pBuffer[i] = value;
-            }
+            Unsafe.InitBlock(((byte*)_pBuffer) + index, value, (uint)length);
```

Comment:
initblk assumes that the starting address is aligned to the natural size of the machine. The operation of the initblk instructions can be altered by an immediately preceding Volatile or Unaligned prefix instruction.

Does Unsafe.InitBlock use such prefix?